Cognite.Simulator.Utils/RoutineLibraryBase.cs

```diff
@@ -215,12 +241,21 @@ private async Task ReadRoutineRevisions(CancellationToken token)
 
             var routineRevisions = await routineRevisionsRes.ToListAsync(token).ConfigureAwait(false);
 
-            foreach (var routineRev in routineRevisions)
+            if (routineRevisions.Any())
             {
-                if (!RoutineRevisions.ContainsKey(routineRev.Id.ToString()))
+                foreach (var routineRev in routineRevisions)
                 {
                     ReadAndSaveRoutineRevision(routineRev);
                 }
+
+                var maxCreatedMs = RoutineRevisions
+                    .Select(s => s.Value.CreatedTime)
+                    .Max();
+
+                var maxCreatedDt = CogniteTime.FromUnixTimeMilliseconds(maxCreatedMs);
+                LibState.UpdateDestinationRange(
```

Comment:
Should we log this as well?

`_logger.LogDebug("Destination range updated to start: {Start}, end: {End}",
    maxCreatedDt, maxCreatedDt);`

Reply:
OK